include/range/v3/utility/tagged_pair.hpp

```diff
@@ -14,175 +15,196 @@
+                    noexcept(std::is_nothrow_move_constructible<Untagged>::value)    \
+                  : Next(detail::move(that))                                         \
+                {}                                                                   \
+                CONCEPT_REQUIRES(CopyConstructible<Untagged>())                      \
+                constexpr getter(Untagged const &that)                               \
+                    noexcept(std::is_nothrow_copy_constructible<Untagged>::value)    \
+                  : Next(that)                                                       \
+                {}                                                                   \
+                getter &operator=(getter &&) = default;                              \
+                getter &operator=(getter const &) = default;                         \
+                RANGES_CXX14_CONSTEXPR                                               \
+                meta::_t<std::tuple_element<I, Untagged>> &NAME() &                  \
+                    noexcept(noexcept(                                               \
+                        detail::adl_get<I>(std::declval<Untagged &>())))             \
+                {                                                                    \
+                    return detail::adl_get<I>(static_cast<Untagged &>(*this));       \
+                }                                                                    \
+                RANGES_CXX14_CONSTEXPR                                               \
+                meta::_t<std::tuple_element<I, Untagged>> &&NAME() &&                \
+                    noexcept(noexcept(                                               \
+                        detail::adl_get<I>(std::declval<Untagged>())))               \
+                {                                                                    \
+                    return detail::adl_get<I>(static_cast<Untagged &&>(*this));      \
+                }                                                                    \
+                constexpr                                                            \
+                meta::_t<std::tuple_element<I, Untagged>> const &NAME() const &      \
+                    noexcept(noexcept(                                               \
+                        detail::adl_get<I>(std::declval<Untagged const &>())))       \
+                {                                                                    \
+                    return detail::adl_get<I>(static_cast<Untagged const &>(*this)); \
+                }                                                                    \
+            };                                                                       \
+        };                                                                           \
+    }                                                                                \
```

Comment:
> Should we update the Ranges TS?

Yes, although I think we will need a better design for `get` as a customization point.